MREUnityRuntime/MREUnityRuntimeLib/Core/ActorCommandQueue.cs

```diff
@@ -0,0 +1,100 @@
+﻿using MixedRealityExtension.App;
+using MixedRealityExtension.Core.Interfaces;
+using MixedRealityExtension.Messaging.Payloads;
+using MixedRealityExtension.Util;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MixedRealityExtension.Core
+{
+    internal class ActorCommandQueue
+    {
+        #region Public Accessors
+
+        public int Count => queue.Count;
+
+        #endregion
+
+        #region Constructor
+
+        public ActorCommandQueue(Guid actorId, MixedRealityExtensionApp app)
+        {
+            this.actorId = actorId;
+            this.app = app;
+        }
+
+        #endregion
+
+        #region Public Methods
+
+        public void Enqueue(NetworkCommandPayload payload, Action onCompleteCallback)
+        {
+            queue.Enqueue(new QueuedCommand()
+            {
+                Payload = payload,
+                OnCompleteCallback = onCompleteCallback
+            });
+        }
+
+        public void Update()
+        {
+            if (busy)
+            {
+                return;
+            }
+
+            if (actor == null)
+            {
+                actor = app.FindActor(actorId) as Actor;
+            }
+
+            if (actor != null && queue.Count > 0)
+            {
+                activeCommand = queue.Dequeue();
+                app.ExecuteCommandPayload(actor, activeCommand.Payload, OnCommandComplete);
```

Comment:
Look at handling exceptions here instead of forcing each handler function to have a try/finally block.  Centralizes our exception handling to one point for the command queue processing.